src/main/java/org/opentripplanner/middleware/utils/ConfigUtils.java

```diff
@@ -53,6 +58,40 @@ public static void loadConfig(String[] args) throws IOException {
             envConfigStream = new FileInputStream(new File(args[0]));
         }
         envConfig = yamlMapper.readTree(envConfigStream);
+        validateConfig();
+    }
+
+    /**
+     * Validate the environment configuration against the environment configuration schema. If the config is not available,
+     * does not match the schema or an exception is thrown, exit the application.
+     */
+    private static void validateConfig() throws IOException {
+        if ("false".equals(getConfigPropertyAsText("VALIDATE_ENVIRONMENT_CONFIG", "true"))) {
+            LOG.warn("Environment configuration schema validation disabled.");
+            return;
+        }
+        if (envConfig == null) {
+            LOG.error("Environment configuration not available to validate!");
+            System.exit(0);
```

Comment:
I think `0` is typically a success/non-error code (maybe we could just use `1` throughout this file). Also, it might be simpler to surround this all in a try/catch and throw an IllegalArgumentException (with each error message you've defined) each place the `exit` method is invoked. Then, in the catch block we could call System#exit once and finally.

Reply:
Updated.